src/main/java/com/kodilla/ecommercee/service/CartService.java

```diff
@@ -0,0 +1,73 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.domain.Cart;
+import com.kodilla.ecommercee.domain.ProductItem;
+import com.kodilla.ecommercee.domain.dto.CartDto;
+import com.kodilla.ecommercee.domain.dto.OrderDto;
+import com.kodilla.ecommercee.mapper.CartMapper;
+import com.kodilla.ecommercee.repository.CartRepository;
+import com.kodilla.ecommercee.repository.OrderRepository;
+import com.kodilla.ecommercee.repository.ProductItemRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class CartService {
+
+    @Autowired
+    private CartRepository cartRepository;
+
+    @Autowired
+    private OrderRepository orderRepository;
+
+    @Autowired
+    private ProductItemRepository productItemRepository;
+
+    @Autowired
+    private CartMapper cartMapper;
+
+    public Cart createCart(@RequestParam long userId) {
+        return new Cart();
+    }
+
+    public Optional<Cart> getCart(final long id) { //ProductItem zamiast Cart
+        return cartRepository.findById(id);
+    }
+
+    public Optional<Cart> addItemToCard(@RequestParam long cartId, @RequestBody ProductItem productItem) {
+        // na podstawie CartId wyciagnac z Repozytorium koszyk
+        Optional<Cart> cartOptional = cartRepository.findById(cartId);
+        if (cartOptional.isPresent()) {
+            Cart cart = cartOptional.get();
+            // dla wyciagnietego koszyka dodac productItem do listy ProductsItems
+            cart.getProductItems().add(productItem);
+
+            cartRepository.save(cart);
+            return Optional.of(cart);
+        }
+        return Optional.empty();
+    }
+
+    public Optional<Cart> removeItemFromCard(@RequestParam long cartId, @RequestParam long itemId) {
+        Optional<Cart> cartOptional = cartRepository.findById(cartId);
+        if (cartOptional.isPresent()) {
```

Comment:
Zastanawiam się nad sytuacją gdy cartOptional będzie nullem wtedy wyrażenie warunkowe cartOptional.isPresent() będzie fałszywe i nie spełni sie resztę logiki. Będzie to prowadzić to różnego rodzaju błędu bo nigdzie nie jesteśmy poinformowani czy obiekt został znaleziony czy nie. Zostanie zwrócony poprostu Optional.empty().

Co powiesz na to aby wyszukiwanie danego elementu wygladało tak : 
Cart cart = cartRepository.findById(cartId).orElseThrow(new CartException())
Wtedy jeżeli nie znajdzie obiektu to rzuci wyjątek że dany element nie został znaleziony, dzięki czemu kod będzie nas lepiej informował co dzieje sie w programie.

I w zasadzie to samo tyczy się linijki 46.

Reply:
Hej, masz rację z tym.